File: leetcode/c++/Palindrome_Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0) return false;
        if(x < 10) return true;
        
        int y = x;
        int a = 1;
        while(y >= 10){
            y /= 10;
            a *= 10;
        }
        
        while(a > 1){
            if(x/a != x%10) return false;
            x %= a;
            x /= 10;
            a /= 100;
        }
        
        return true;
    }
};

File: leetcode/java/PalindromeNumber.java
class Solution {
    public boolean isPalindrome(int x) {
        //if(x == Integer.MIN_VALUE) return false;
        if(x < 0) return false; // depends what we want
        
        //find max base
        int maxBase = findMaxBase(x);
        while(x > 0) {
            if(x/maxBase != x%10) return false;
            x %=maxBase;
            x /=10;
            maxBase /= 100;
        }
        return true;
    }
    
    private int findMaxBase(int x) {
        int base = 1;
        while(x >= 10){
            base *=10;
            x /= 10;
        }
        return base;
    }
}
